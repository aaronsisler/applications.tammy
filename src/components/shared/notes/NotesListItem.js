import React from 'react';
import PropTypes from 'prop-types';

export default class NotesListItem extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div className="notes_list_item">
                <div className="notes_list_item__status_note">
                    <textarea
                        className="textarea"
                        cols="100"
                        readOnly={true}
                        rows="5"
                        value={this.props.statusNote}
                    >
                    </textarea>
                </div>
            </div>
        );
    }
}

NotesListItem.propTypes = {
    statusNote: PropTypes.string.isRequired,
};
