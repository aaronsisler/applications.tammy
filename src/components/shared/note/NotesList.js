import React from 'react';
import PropTypes from 'prop-types';
import NotesListItem from 'Shared/note/NotesListItem';

export default class NotesList extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        if (this.props.notes.length === 0) {
            return (
                <div id="notes_list" className="empty">
                    No available notes
                </div>
            );
        }

        return (
            <div id="notes_list">
                {
                    this.props.notes.map((note, index) =>
                        <NotesListItem
                            key={index}
                            {...note}
                        />)
                }
            </div>
        );
    }
}

NotesList.propTypes = {
    notes: PropTypes.array,
};
