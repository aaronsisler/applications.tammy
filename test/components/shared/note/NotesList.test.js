import React from 'react';
import { shallow } from 'enzyme';
import NotesList from 'Shared/note/NotesList';
import notes from '../../../fixtures/notes';

describe('NotesList', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <NotesList
                notes={notes}
            />
        );
    })

    it('should render NotesList correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
