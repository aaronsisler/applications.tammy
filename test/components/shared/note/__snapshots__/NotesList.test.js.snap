// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesList should render NotesList correctly 1`] = `
<div
  className="notes_list"
>
  <NotesListItem
    key="0"
    noteMessage="First note"
  />
  <NotesListItem
    key="1"
    noteMessage="Second note"
  />
</div>
`;
